src/quartapp/chat.py

```diff
@@ -17,26 +24,52 @@
 
 @bp.before_app_serving
 async def configure_openai():
-    # Authenticate using the default Azure credential chain
-    # See https://docs.microsoft.com/azure/developer/python/azure-sdk-authenticate#defaultazurecredential
-    # This will *not* work inside a local Docker container.
-    # If using managed user-assigned identity, make sure that AZURE_CLIENT_ID is set
-    # to the client ID of the user-assigned identity.
-    current_app.logger.info("Using Azure OpenAI with default credential")
-    default_credential = azure.identity.aio.DefaultAzureCredential(exclude_shared_token_cache_credential=True)
-    token_provider = azure.identity.aio.get_bearer_token_provider(
-        default_credential, "https://cognitiveservices.azure.com/.default"
+    # Configure Environment Variables
+    AZURE_TENANT_ID = os.getenv("AZURE_TENANT_ID")
+    AZURE_CLIENT_ID = os.getenv("AZURE_CLIENT_ID")
+
+    # Check if AZURE_CLIENT_ID is set
+    if not AZURE_CLIENT_ID:
+        raise ValueError("AZURE_CLIENT_ID is required for Authentication.") 
+    
+    # Check if AZURE_TENANT_ID is set, 
+    # If not set, use AzureDeveloperCliCredential with the default tenant.
+    if not AZURE_TENANT_ID:
+        azure_developer_cli_credential = AzureDeveloperCliCredential(tenant_id=AZURE_TENANT_ID, process_timeout=60)
+    else:
+        azure_developer_cli_credential = AzureDeveloperCliCredential(process_timeout=60)
+
+    # Create a ChainedTokenCredential with ManagedIdentityCredential and AzureDeveloperCliCredential
+    #  - ManagedIdentityCredential is used for Azure Functions and Azure App Service
+    #       This is a user-assigned managed identity. User-assigned managed identities are supported by passing the client_id to ManagedIdentityCredential

```

Comment:
```suggestion
    #       This uses a user-assigned managed identity. User-assigned managed identities are supported by passing the client_id to ManagedIdentityCredential

```

Am confused as to what "this" is in that sentence

Reply:
I think that sentence is unnecessary. I think it's better to remove it.